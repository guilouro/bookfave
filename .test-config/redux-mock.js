import React from 'react'
import { createStore, combineReducers, applyMiddleware } from 'redux'
import { render } from 'react-testing-library'
import { Provider } from 'react-redux'

export const mockStore = mock => {
  const reducers = {}

  Object.keys(mock).forEach(key => {
    reducers[key] = () => mock[key]
  })

  return createStore(combineReducers(reducers))
}

export const renderWithRedux = (component, store) =>
  render(<Provider store={store}>{component}</Provider>)
