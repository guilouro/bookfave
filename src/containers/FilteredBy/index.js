import React from 'react'
import PropTypes from 'prop-types'
import Tag from '../../components/Tag'
import { connect } from 'react-redux'
import { filterByTag } from '../../actions/favorites'

const FilteredBy = ({ tagName, clearFilter }) =>
  !!tagName && (
    <p>
      Filtered by: <Tag name={tagName} onClose={clearFilter} />
    </p>
  )

FilteredBy.propTypes = {
  clearFilter: PropTypes.func.isRequired,
  tagName: PropTypes.string
}

FilteredBy.defaultProps = {
  tagName: ''
}

const mapStateToProps = ({ favorites }) => ({
  tagName: favorites.filterTag
})

const mapDispatchToProps = dispatch => ({
  clearFilter: () => dispatch(filterByTag(''))
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(FilteredBy)
