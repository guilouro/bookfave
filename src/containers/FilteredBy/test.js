import React from 'react'
import { cleanup } from 'react-testing-library'
import FilteredBy from '.'
import { mockStore, renderWithRedux } from '../../../.test-config/redux-mock'

describe('<FilteredBy />', () => {
  afterEach(cleanup)
  describe('without tagName', () => {
    const store = mockStore({
      favorites: {
        filterTag: '',
        entities: {},
        ids: []
      }
    })

    it('Should render correctly', () => {
      const { container } = renderWithRedux(<FilteredBy />, store)

      expect(container.firstChild).toBeNull()
    })
  })

  describe('with tagName', () => {
    const store = mockStore({
      favorites: {
        filterTag: 'wallet',
        entities: {},
        ids: []
      }
    })

    it('Should render correctly', () => {
      const { container, queryByText } = renderWithRedux(<FilteredBy />, store)

      expect(queryByText(/wallet/g)).toBeTruthy()
      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
