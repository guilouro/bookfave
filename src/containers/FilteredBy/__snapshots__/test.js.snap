// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilteredBy /> with tagName Should render correctly 1`] = `
<p>
  Filtered by: 
  <span
    class="sc-bdVaJa eeVrHu"
  >
    <span>
      wallet
    </span>
    <button
      class="sc-bwzfXH sc-htpNat chgZfo"
    >
      âœ•
    </button>
  </span>
</p>
`;
